In [5]:
import pandas as pd
from pytrends.request import TrendReq

In [4]:
!pip install pytrends

  Obtaining dependency information for pytrends from https://files.pythonhosted.org/packages/68/ba/7a24a3723c790000faf880505ff1cc46f4d29f46dd353037938a070c4d23/pytrends-4.9.2-py3-none-any.whl.metadata


In [6]:
# Initialize Pytrends
pytrends = TrendReq(hl='es-PE')  # Set the region to Peru

In [3]:
def chunks(lst, n):
    """Yield successive n-sized chunks from lst."""
    for i in range(0, len(lst), n):
        yield lst[i:i + n]

In [7]:
df= pd.read_csv('models.csv')
#show df dimensions

print(df.shape)

(5154, 4)


In [12]:
#give me a list of unike makes
makes = df['make'].unique().tolist()

In [13]:
makes

['chevrolet',
 'ford',
 'honda',
 'hyundai',
 'mazda',
 'mitsubishi',
 'nissan',
 'subaru',
 'suzuki',
 'toyota',
 'volkswagen',
 'kia']

In [8]:
# get make and model from the dataframe and store it in a list
models = df[['make', 'model']].values.tolist()
#join in a single string
models = [' '.join(make_model) for make_model in models]
#print models dimensions
print(len(models))

5154


In [38]:
models

['chevrolet 1500 Extended Cab',
 'chevrolet 1500 Regular Cab',
 'chevrolet 2500 Extended Cab',
 'chevrolet 2500 Regular Cab',
 'chevrolet 3500 Crew Cab',
 'chevrolet 3500 Extended Cab',
 'chevrolet 3500 Regular Cab',
 'chevrolet APV Cargo',
 'chevrolet Astro Cargo',
 'chevrolet Astro Passenger',
 'chevrolet Beretta',
 'chevrolet Blazer',
 'chevrolet Camaro',
 'chevrolet Caprice',
 'chevrolet Cavalier',
 'chevrolet Corsica',
 'chevrolet Corvette',
 'chevrolet G-Series G10',
 'chevrolet G-Series G20',
 'chevrolet G-Series G30',
 'chevrolet Lumina',
 'chevrolet Lumina APV',
 'chevrolet S10 Blazer',
 'chevrolet S10 Extended Cab',
 'chevrolet S10 Regular Cab',
 'chevrolet Sportvan G10',
 'chevrolet Sportvan G20',
 'chevrolet Sportvan G30',
 'chevrolet Suburban 1500',
 'chevrolet Suburban 2500',
 'ford Aerostar Cargo',
 'ford Aerostar Passenger',
 'ford Bronco',
 'ford Club Wagon',
 'ford Crown Victoria',
 'ford Econoline E150 Cargo',
 'ford Econoline E250 Cargo',
 'ford Econoline E350 Cargo

In [40]:
#search hyundai accent and give me the index
print(models.index('hyundai Accent'))

427


In [51]:
models[102]

'toyota Corolla'

427 'hyundai Accent' 31 tiene repeticiones xD
102 toyota Corolla 65

In [9]:
# create a new list from models, take the models[102] and put it in the first position

new_models = [models[102]] + models[:102] + models[102:]
new_models

['toyota Corolla',
 'chevrolet 1500 Extended Cab',
 'chevrolet 1500 Regular Cab',
 'chevrolet 2500 Extended Cab',
 'chevrolet 2500 Regular Cab',
 'chevrolet 3500 Crew Cab',
 'chevrolet 3500 Extended Cab',
 'chevrolet 3500 Regular Cab',
 'chevrolet APV Cargo',
 'chevrolet Astro Cargo',
 'chevrolet Astro Passenger',
 'chevrolet Beretta',
 'chevrolet Blazer',
 'chevrolet Camaro',
 'chevrolet Caprice',
 'chevrolet Cavalier',
 'chevrolet Corsica',
 'chevrolet Corvette',
 'chevrolet G-Series G10',
 'chevrolet G-Series G20',
 'chevrolet G-Series G30',
 'chevrolet Lumina',
 'chevrolet Lumina APV',
 'chevrolet S10 Blazer',
 'chevrolet S10 Extended Cab',
 'chevrolet S10 Regular Cab',
 'chevrolet Sportvan G10',
 'chevrolet Sportvan G20',
 'chevrolet Sportvan G30',
 'chevrolet Suburban 1500',
 'chevrolet Suburban 2500',
 'ford Aerostar Cargo',
 'ford Aerostar Passenger',
 'ford Bronco',
 'ford Club Wagon',
 'ford Crown Victoria',
 'ford Econoline E150 Cargo',
 'ford Econoline E250 Cargo',
 'ford E

In [10]:
# delete repeated values
new_models = list(dict.fromkeys(new_models))
new_models

['toyota Corolla',
 'chevrolet 1500 Extended Cab',
 'chevrolet 1500 Regular Cab',
 'chevrolet 2500 Extended Cab',
 'chevrolet 2500 Regular Cab',
 'chevrolet 3500 Crew Cab',
 'chevrolet 3500 Extended Cab',
 'chevrolet 3500 Regular Cab',
 'chevrolet APV Cargo',
 'chevrolet Astro Cargo',
 'chevrolet Astro Passenger',
 'chevrolet Beretta',
 'chevrolet Blazer',
 'chevrolet Camaro',
 'chevrolet Caprice',
 'chevrolet Cavalier',
 'chevrolet Corsica',
 'chevrolet Corvette',
 'chevrolet G-Series G10',
 'chevrolet G-Series G20',
 'chevrolet G-Series G30',
 'chevrolet Lumina',
 'chevrolet Lumina APV',
 'chevrolet S10 Blazer',
 'chevrolet S10 Extended Cab',
 'chevrolet S10 Regular Cab',
 'chevrolet Sportvan G10',
 'chevrolet Sportvan G20',
 'chevrolet Sportvan G30',
 'chevrolet Suburban 1500',
 'chevrolet Suburban 2500',
 'ford Aerostar Cargo',
 'ford Aerostar Passenger',
 'ford Bronco',
 'ford Club Wagon',
 'ford Crown Victoria',
 'ford Econoline E150 Cargo',
 'ford Econoline E250 Cargo',
 'ford E

In [8]:
print(len(new_models))

574


In [52]:

'''
pytrends.build_payload([models[102]], cat=0, timeframe='today 3-m', geo='PE')
data = pytrends.interest_over_time()
mean = data.mean()
print(mean)
'''

toyota Corolla    65.623656
isPartial          0.010753
dtype: float64


In [71]:
for chunk in [ lst + [new_models[0]] for lst in list(chunks(new_models[1:], 4))]:
    print(chunk)

['chevrolet 1500 Extended Cab', 'chevrolet 1500 Regular Cab', 'chevrolet 2500 Extended Cab', 'chevrolet 2500 Regular Cab', 'toyota Corolla']
['chevrolet 3500 Crew Cab', 'chevrolet 3500 Extended Cab', 'chevrolet 3500 Regular Cab', 'chevrolet APV Cargo', 'toyota Corolla']
['chevrolet Astro Cargo', 'chevrolet Astro Passenger', 'chevrolet Beretta', 'chevrolet Blazer', 'toyota Corolla']
['chevrolet Camaro', 'chevrolet Caprice', 'chevrolet Cavalier', 'chevrolet Corsica', 'toyota Corolla']
['chevrolet Corvette', 'chevrolet G-Series G10', 'chevrolet G-Series G20', 'chevrolet G-Series G30', 'toyota Corolla']
['chevrolet Lumina', 'chevrolet Lumina APV', 'chevrolet S10 Blazer', 'chevrolet S10 Extended Cab', 'toyota Corolla']
['chevrolet S10 Regular Cab', 'chevrolet Sportvan G10', 'chevrolet Sportvan G20', 'chevrolet Sportvan G30', 'toyota Corolla']
['chevrolet Suburban 1500', 'chevrolet Suburban 2500', 'ford Aerostar Cargo', 'ford Aerostar Passenger', 'toyota Corolla']
['ford Bronco', 'ford Club 

In [72]:
for chunk_number, chunk in enumerate([lst + [new_models[0]] for lst in list(chunks(new_models[1:], 4))]):
    print(chunk_number)
    print(chunk)

0
['chevrolet 1500 Extended Cab', 'chevrolet 1500 Regular Cab', 'chevrolet 2500 Extended Cab', 'chevrolet 2500 Regular Cab', 'toyota Corolla']
1
['chevrolet 3500 Crew Cab', 'chevrolet 3500 Extended Cab', 'chevrolet 3500 Regular Cab', 'chevrolet APV Cargo', 'toyota Corolla']
2
['chevrolet Astro Cargo', 'chevrolet Astro Passenger', 'chevrolet Beretta', 'chevrolet Blazer', 'toyota Corolla']
3
['chevrolet Camaro', 'chevrolet Caprice', 'chevrolet Cavalier', 'chevrolet Corsica', 'toyota Corolla']
4
['chevrolet Corvette', 'chevrolet G-Series G10', 'chevrolet G-Series G20', 'chevrolet G-Series G30', 'toyota Corolla']
5
['chevrolet Lumina', 'chevrolet Lumina APV', 'chevrolet S10 Blazer', 'chevrolet S10 Extended Cab', 'toyota Corolla']
6
['chevrolet S10 Regular Cab', 'chevrolet Sportvan G10', 'chevrolet Sportvan G20', 'chevrolet Sportvan G30', 'toyota Corolla']
7
['chevrolet Suburban 1500', 'chevrolet Suburban 2500', 'ford Aerostar Cargo', 'ford Aerostar Passenger', 'toyota Corolla']
8
['ford Br

In [60]:
for chunk in [ lst + [new_models[0]] for lst in list(chunks(new_models[1:], 4))]:
    pytrends.build_payload(chunk, cat=0, timeframe='today 3-m', geo='PE')
    data = pytrends.interest_over_time()
    mean = data.mean()
    print(mean)
    break  

chevrolet 1500 Extended Cab     0.000000
chevrolet 1500 Regular Cab      0.419355
chevrolet 2500 Extended Cab     0.000000
chevrolet 2500 Regular Cab      0.430108
toyota Corolla                 65.462366
isPartial                       0.010753
dtype: float64


In [69]:
data = []
# Process each chunk
for chunk_number, lst in enumerate([lst + [new_models[0]] for lst in list(chunks(new_models[1:], 4))], start=1):
    is_the_same_value = all(model == lst[0] for model in lst)  # Check if all models are the same
    data.append([chunk_number, is_the_same_value, 0.0])

# Create the dataframe
df = pd.DataFrame(data, columns=['chunk_number', 'is_the_same_value', 'interest'])

# Display the dataframe
print(df)

     chunk_number  is_the_same_value  interest
0               1              False       0.0
1               2              False       0.0
2               3              False       0.0
3               4              False       0.0
4               5              False       0.0
..            ...                ...       ...
139           140              False       0.0
140           141              False       0.0
141           142              False       0.0
142           143              False       0.0
143           144              False       0.0

[144 rows x 3 columns]


In [70]:
df[df['is_the_same_value'] == True]

Empty DataFrame
Columns: [chunk_number, is_the_same_value, interest]
Index: []

## V1: falta asignar el 0.0 por chunks

In [74]:
def calculate_means(chunk):
    pytrends.build_payload(chunk, cat=0, timeframe='today 5-y', geo='PE')
    data = pytrends.interest_over_time()
    return data.mean()

In [ ]:
def chunks(lst, n):
    """Yield successive n-sized chunks from lst."""
    for i in range(0, len(lst), n):
        yield lst[i:i + n]

In [75]:
data = []

for chunk_number, chunk in enumerate([lst + [new_models[0]] for lst in list(chunks(new_models[1:], 4))]):
    means = calculate_means(chunk)
    for i, model in enumerate(chunk):
        is_the_same_value = model == 'toyota Corolla'
        data.append([chunk_number, is_the_same_value, model, means[i]])
    break
    

# Create the DataFrame
df = pd.DataFrame(data, columns=['chunk_number', 'is_the_same_value', 'model', 'value'])

# Display the DataFrame
print(df)

   chunk_number  is_the_same_value                        model      value
0             0              False  chevrolet 1500 Extended Cab   0.000000
1             0              False   chevrolet 1500 Regular Cab   0.000000
2             0              False  chevrolet 2500 Extended Cab   0.011450
3             0              False   chevrolet 2500 Regular Cab   0.038168
4             0               True               toyota Corolla  61.958015


C:\Users\Usuario\AppData\Local\Temp\ipykernel_53604\1371924532.py:7: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  data.append([chunk_number, is_the_same_value, model, means[i]])


## V2 con try catch

In [11]:
import pandas as pd
import time

# Initialize variables
data = []
previous_df = pd.DataFrame()

def calculate_means(chunk):
    pytrends.build_payload(chunk, cat=0, timeframe='today 5-y', geo='PE')
    data = pytrends.interest_over_time()
    return data.mean()

def chunks(lst, n):
    """Yield successive n-sized chunks from lst."""
    for i in range(0, len(lst), n):
        yield lst[i:i + n]

for chunk_number, chunk in enumerate([lst + [new_models[0]] for lst in list(chunks(new_models[1:], 4))]):
    if(chunk_number < 83): continue
    try:
        means = calculate_means(chunk)

        # Display relevant details in the console
        print(f"Processing chunk {chunk_number}")
        print("Chunk models:", chunk)

        # Save the current chunk's data
        current_data = []
        for i, model in enumerate(chunk):
            is_the_same_value = model == 'toyota Corolla'
            current_data.append([chunk_number, is_the_same_value, model, means[i]])

        current_df = pd.DataFrame(current_data, columns=['chunk_number', 'is_the_same_value', 'model', 'value'])

        # Save current and previous dataframes to Excel
        with pd.ExcelWriter(f'chunk_data_{chunk_number}.xlsx') as writer:
            current_df.to_excel(writer, sheet_name='Current', index=False)
            previous_df.to_excel(writer, sheet_name='Previous', index=False)

        # Update the previous_df to be the current one for the next iteration
        previous_df = current_df.copy()

        time.sleep(60)

    except Exception as e:
        print(f"Error processing chunk {chunk_number}: {e}")
        break


Processing chunk 83
Chunk models: ['suzuki XL7', 'toyota FJ Cruiser', 'toyota Tundra CrewMax', 'toyota Yaris', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 84
Chunk models: ['volkswagen Eos', 'chevrolet Malibu (Classic)', 'ford E150 Cargo', 'ford E250 Cargo', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 85
Chunk models: ['ford F450 Super Duty Crew Cab', 'ford Taurus X', 'nissan Rogue', 'subaru Tribeca', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 86
Chunk models: ['volkswagen GLI', 'volkswagen Touareg 2', 'chevrolet Traverse', 'ford Flex', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 87
Chunk models: ['hyundai Genesis', 'kia Borrego', 'nissan 370Z', 'nissan cube', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 88
Chunk models: ['nissan GT-R', 'suzuki Equator Crew Cab', 'suzuki Equator Extended Cab', 'toyota Venza', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 89
Chunk models: ['volkswagen CC', 'volkswagen Routan', 'volkswagen Tiguan', 'ford Transit Connect Cargo', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 90
Chunk models: ['ford Transit Connect Passenger', 'honda Accord Crosstour', 'hyundai Genesis Coupe', 'kia Forte', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 91
Chunk models: ['kia Soul', 'suzuki Kizashi', 'chevrolet Cruze', 'chevrolet Volt', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 92
Chunk models: ['ford Fiesta', 'honda CR-Z', 'hyundai Equus', 'mazda MAZDA2', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 93
Chunk models: ['mitsubishi Outlander Sport', 'nissan JUKE', 'nissan LEAF', 'chevrolet Captiva Sport', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 94
Chunk models: ['chevrolet Sonic', 'honda Crosstour', 'hyundai Veloster', 'mitsubishi i-MiEV', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 95
Chunk models: ['nissan NV1500 Cargo', 'nissan NV2500 HD Cargo', 'nissan NV3500 HD Cargo', 'nissan NV3500 HD Passenger', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 96
Chunk models: ['toyota Prius c', 'toyota Prius Plug-in Hybrid', 'toyota Prius v', 'volkswagen Beetle', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 97
Chunk models: ['volkswagen Jetta SportWagen', 'chevrolet Spark', 'ford C-MAX Energi', 'ford C-MAX Hybrid', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 98
Chunk models: ['ford Focus ST', 'ford Fusion Energi', 'hyundai Santa Fe Sport', 'mazda CX-5', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 99
Chunk models: ['nissan NV200', 'subaru BRZ', 'subaru XV Crosstrek', 'chevrolet Impala Limited', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 100
Chunk models: ['chevrolet Silverado 1500 Double Cab', 'chevrolet Spark EV', 'chevrolet SS', 'honda Accord Hybrid', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 101
Chunk models: ['kia Cadenza', 'nissan Rogue Select', 'chevrolet City Express', 'chevrolet Silverado 2500 HD Double Cab', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 102
Chunk models: ['chevrolet Silverado 3500 HD Double Cab', 'chevrolet Suburban', 'chevrolet Trax', 'ford Transit 150 Van', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 103
Chunk models: ['ford Transit 150 Wagon', 'ford Transit 250 Van', 'ford Transit 350 HD Van', 'ford Transit 350 Van', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 104
Chunk models: ['ford Transit 350 Wagon', 'hyundai Elantra GT', 'hyundai Tucson Fuel Cell', 'kia Forte Koup', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 105
Chunk models: ['kia Forte5', 'kia K900', 'kia Optima Hybrid', 'kia Soul EV', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 106
Chunk models: ['mitsubishi Lancer Evolution', 'nissan NV200 Taxi', 'subaru WRX', 'volkswagen e-Golf', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 107
Chunk models: ['volkswagen Golf GTI', 'volkswagen Golf R', 'volkswagen Golf SportWagen', 'chevrolet Cruze Limited', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 108
Chunk models: ['chevrolet Malibu Limited', 'chevrolet Suburban 3500HD', 'honda HR-V', 'hyundai Sonata Hybrid', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 109
Chunk models: ['hyundai Sonata Plug-in Hybrid', 'mazda CX-3', 'nissan TITAN XD Crew Cab', 'nissan Versa Note', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 110
Chunk models: ['subaru Crosstrek', 'toyota Mirai', 'chevrolet Bolt EV', 'honda Civic Type R', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 111
Chunk models: ['hyundai Ioniq Electric', 'hyundai Ioniq Hybrid', 'kia Niro', 'kia Optima Plug-in Hybrid', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 112
Chunk models: ['mazda MX-5 Miata RF', 'mitsubishi Mirage G4', 'nissan Rogue Sport', 'nissan TITAN Single Cab', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 113
Chunk models: ['nissan TITAN XD King Cab', 'nissan TITAN XD Single Cab', 'toyota 86', 'toyota Avalon Hybrid', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 114
Chunk models: ['toyota Camry Hybrid', 'toyota Corolla iM', 'toyota Highlander Hybrid', 'toyota Prius Prime', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 115
Chunk models: ['toyota RAV4 Hybrid', 'toyota Yaris iA', 'volkswagen Golf Alltrack', 'volkswagen Tiguan Limited', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 116
Chunk models: ['ford EcoSport', 'ford Expedition MAX', 'honda Clarity Plug-in Hybrid', 'hyundai Ioniq Plug-in Hybrid', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 117
Chunk models: ['hyundai Kona', 'kia Niro Plug-in Hybrid', 'kia Stinger', 'mitsubishi Eclipse Cross', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 118
Chunk models: ['mitsubishi Outlander PHEV', 'nissan Kicks', 'toyota C-HR', 'volkswagen Atlas', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 119
Chunk models: ['chevrolet Silverado 1500 LD Double Cab', 'ford F450 Super Duty Regular Cab', 'ford Ranger SuperCab', 'ford Ranger SuperCrew', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 120
Chunk models: ['honda Clarity Electric', 'honda Clarity Fuel Cell', 'hyundai Kona Electric', 'hyundai NEXO', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 121
Chunk models: ['hyundai Santa Fe XL', 'kia Niro EV', 'subaru Ascent', 'toyota Corolla Hatchback', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Processing chunk 122
Chunk models: ['volkswagen Arteon', 'volkswagen Jetta GLI', 'ford Fusion Plug-in Hybrid', 'ford Transit 250 Cargo Van', 'toyota Corolla']


C:\Users\Usuario\AppData\Local\Temp\ipykernel_45704\3699134737.py:31: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  current_data.append([chunk_number, is_the_same_value, model, means[i]])


Error processing chunk 123: The request failed: Google returned a response with code 429


Error processing chunk 82: The request failed: Google returned a response with code 429


Error processing chunk 84: The request failed: Google returned a response with code 429


Error processing chunk 123: The request failed: Google returned a response with code 429

## Nueva parte

In [14]:
modelos_faltantes = ['changan X7 Plus', 'changan UNI-K', 'changan UNI-T', 'changan New CS55 Plus', 'changan New CS15', 'changan New CS35 Plus', 'changan New Van', 'changan HONOR S', 'changan Grand SUPERVAN', 'changan Grand VAN Turismo', 'changan New Alsvin', 'changan NEW F70',
'dfsk Glory 500', 'dfsk Glory iX5', 'dfsk DFSK E5', 'dfsk Glory 600', 'dfsk Glory 560', 'dfsk Glory 580', 'dfsk Glory 330', 'dfsk C37', 'dfsk K07S', 'dfsk K05S Cargo Panel', 'dfsk C31 Cabina Simple', 'dfsk C35 Cargo Panel',
'jac JS2', 'jac JS3', 'jac JS4', 'jac JS6', 'jac JS8', 'jac T6', 'jac T8', 'jac T8 PRO', 'jac Nueva Refine', 'jac Sunray', 'jac X200', 'jac E-JS1', 'jac E-JS4', 'jac E-T8', 'jac E-M3',
'renault New Kwid', 'renault Stepway', 'renault All New Duster', 'renault New Renault Kardian', 'renault New Koleos', 'renault New Kwid E Tech', 'renault New Oroch', 'renault Master Furgón', 'renault Master Minibus',
'chery Tiggo 8 Pro Hybrid', 'chery Tiggo 8 Pro Max', 'chery Tiggo 8 Pro', 'chery Tiggo 8', 'chery Tiggo 7 Pro Hybrid', 'chery Tiggo 7 Pro', 'chery Tiggo 4 Pro', 'chery Tiggo 2 Pro', 'chery Arrizo 5',
'jetour X70', 'jetour X70Plus', 'jetour Dashing', 'jetour X90Plus', 'jetour Traveller T2',
'geely GX3 Pro', 'geely New Coolray', 'geely New Okavango', 'geely New Starray', 'geely Emgrand',
'great wall H6 GT', 'great wall DARGO', 'great wall H6', 'great wall JOLION', 'great wall POER AT', 'great wall POER MT', 'great wall WINGLE 7 DIESEL', 'great wall WINGLE 5 DIESEL', 'great wall WINGLE 5 GASOLINA', 'great wall H6 HEV', 'great wall JOLION HEV',
'mg ZX', 'mg ZX PLUS', 'mg HS', 'mg RX5', 'mg ONE', 'mg RX8', 'mg 5', 'mg 6', 'mg GT', 'mg 4', 'mg 3', 'mg 3 HYBRID+',
'jinbei F50', 'jinbei H2L', 'jinbei Haise', 'jinbei Konect',
'foton K0', 'foton K1', 'foton K2', 'foton BigVan Toano', 'foton Tunland G7 4x4', 'foton Tunland G7 4x2', 'foton Tunland G9 4x4', 'foton Tunland G9 4x4', 'foton Tunland G9 DS']



In [16]:
new_list = [new_models[0]] + modelos_faltantes
new_list

['toyota Corolla',
 'changan X7 Plus',
 'changan UNI-K',
 'changan UNI-T',
 'changan New CS55 Plus',
 'changan New CS15',
 'changan New CS35 Plus',
 'changan New Van',
 'changan HONOR S',
 'changan Grand SUPERVAN',
 'changan Grand VAN Turismo',
 'changan New Alsvin',
 'changan NEW F70',
 'dfsk Glory 500',
 'dfsk Glory iX5',
 'dfsk DFSK E5',
 'dfsk Glory 600',
 'dfsk Glory 560',
 'dfsk Glory 580',
 'dfsk Glory 330',
 'dfsk C37',
 'dfsk K07S',
 'dfsk K05S Cargo Panel',
 'dfsk C31 Cabina Simple',
 'dfsk C35 Cargo Panel',
 'jac JS2',
 'jac JS3',
 'jac JS4',
 'jac JS6',
 'jac JS8',
 'jac T6',
 'jac T8',
 'jac T8 PRO',
 'jac Nueva Refine',
 'jac Sunray',
 'jac X200',
 'jac E-JS1',
 'jac E-JS4',
 'jac E-T8',
 'jac E-M3',
 'renault New Kwid',
 'renault Stepway',
 'renault All New Duster',
 'renault New Renault Kardian',
 'renault New Koleos',
 'renault New Kwid E Tech',
 'renault New Oroch',
 'renault Master Furgón',
 'renault Master Minibus',
 'chery Tiggo 8 Pro Hybrid',
 'chery Tiggo 8 Pro M

In [ ]:
import pandas as pd
import time

# Initialize variables
data = []
previous_df = pd.DataFrame()

def calculate_means(chunk):
    pytrends.build_payload(chunk, cat=0, timeframe='today 5-y', geo='PE')
    data = pytrends.interest_over_time()
    return data.mean()

def chunks(lst, n):
    """Yield successive n-sized chunks from lst."""
    for i in range(0, len(lst), n):
        yield lst[i:i + n]

for chunk_number, chunk in enumerate([lst + [new_list[0]] for lst in list(chunks(new_list[1:], 4))]):
    if(chunk_number < 83): continue
    try:
        means = calculate_means(chunk)

        # Display relevant details in the console
        print(f"Processing chunk {chunk_number}")
        print("Chunk models:", chunk)

        # Save the current chunk's data
        current_data = []
        for i, model in enumerate(chunk):
            is_the_same_value = model == 'toyota Corolla'
            current_data.append([chunk_number, is_the_same_value, model, means[i]])

        current_df = pd.DataFrame(current_data, columns=['chunk_number', 'is_the_same_value', 'model', 'value'])

        # Save current and previous dataframes to Excel
        with pd.ExcelWriter(f'chunk_data_rest_{chunk_number}.xlsx') as writer:
            current_df.to_excel(writer, sheet_name='Current', index=False)
            previous_df.to_excel(writer, sheet_name='Previous', index=False)

        # Update the previous_df to be the current one for the next iteration
        previous_df = current_df.copy()

        time.sleep(60)

    except Exception as e:
        print(f"Error processing chunk {chunk_number}: {e}")
        break
